frontend/public/components/secrets/create-secret.tsx

```diff
@@ -144,6 +163,308 @@ const withSecretForm = (SubForm) => class SecretFormComponent extends React.Comp
   }
 };
 
+const getImageSecretKey = (secretType) => {
+  switch (secretType) {
+    case 'kubernetes.io/dockercfg':
+      return '.dockercfg';
+    case 'kubernetes.io/dockerconfigjson':
+      return '.dockerconfigjson';
+    default:
+      return secretType;
+  }
+};
+
+const getImageSecretType = (secretKey) => {
+  switch (secretKey) {
+    case '.dockercfg':
+      return 'kubernetes.io/dockercfg';
+    case '.dockerconfigjson':
+      return 'kubernetes.io/dockerconfigjson';
+    default:
+      return secretKey;
+  }
+};
+
+class ImageSecretForm extends React.Component<ImageSecretFormProps, ImageSecretFormState> {
+  constructor(props) {
+    super(props);
+    const data = this.props.isCreate ? {'.dockerconfigjson': '{}'} : this.props.stringData;
+    this.state = {
+      type: this.props.secretType,
+      dataKey: getImageSecretKey(this.props.secretType),
+      stringData: _.mapValues(data, JSON.parse),
+      uploadConfig: 'false',
+    };
+
+    this.onDataChanged = this.onDataChanged.bind(this);
+    this.changeFormType = this.changeFormType.bind(this);
+  }
+  onDataChanged (secretData) {
+    this.setState({
+      stringData: {[this.state.dataKey]: secretData}
+    }, () => this.props.onChange({
+      stringData: _.mapValues(this.state.stringData, JSON.stringify),
```

Comment:
Hmm not sure if an exception can occure here, since the underlying components - `CreateConfigSubform` and `UploadConfigSubform` should take care of it.

In `CreateConfigSubform` when an error occurs over parsing a JSON, the editor should disable the `Save` button (at least thats the logic Im addressing in the upcoming commit)
![1](https://user-images.githubusercontent.com/1668218/43393530-cf4c07e4-93f7-11e8-9782-dea528ec1d43.png)


For the `UploadConfigSubform` - upon secret creation, when the user pastes/enters a non-JSON format to the text-area  he will be notified with a warning:
![2](https://user-images.githubusercontent.com/1668218/43393532-d17b034e-93f7-11e8-89ec-0b41d8e194b0.png)
Here the question is, whether we want to disable the 'Create' button or have it still clickable but don't propagate the changes to the upper components (as the current code is doing).

Reply:
I need to look at this more closely again, but generally speaking: as much as possible, I'd like the console to be a 
"dumb" client that just shows the value and sends what the user types in and lets the API server do any validation / use the data. Problems then are communicated to the user through either API errors on save or events and conditions later on.

Specifically in this case, do we know that this is always going to be a JSON file? What if someday YAML is accepted as a config file format?

I want to be careful of starting down the road of reimplementing logic from API server. We went too far in that direction in the last console, and it causes problems as the API server changes and implementations drift. Even if it works perfectly now, too many assumptions about the format can cause bugs later on.

It might make sense to separate the cases where we create the config file based on individual inputs vs editing an existing value or handling a file upload. If it's not a file we created, we just treat it as a blob.

Reply:
Having a 'dumb' client makes sense together with communicating the errors through API.
But this case is slightly different. If we take the issue of creating the pull/push secret using CLI (only other user access point, besides UI console), there are two ways how to create a pull/push secret:
1. using `oc create secret docker-registry`, where user has to specify - username & pw & email + server(optional). This call will create only the new `k8s.io/dockerconfigjson` secret type.

2. using `oc create secret generic pull --from-file=.dockercfg=../origin/config.json --type=kubernetes.io/dockercfg` which user has to feed with config file (dockercfg or dockerconfigjson format) but he also has to specify the right type of the format.

The first case we use when user fills the form with the server& username & pw & email inputs.

The second one is a bit tricky cause here the user has to know the format type, to help the CLI with creating the right secret type, since its not parsing the config file to find out whats the type. Here I wasnt sure how to do it, without parsing the file.

One thing that comes to my mind is to have a additional selectbox above the config file droparea where user can specify the secret type, `k8s.io/dockercofigjson` by default. That case, the experience would be similar to the one from the CLI, and we could threat the file input as a blob and just wait for the API server response, if something is wrong. This way we can skip any kind of validation.
Thoughts ?